File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.py
n = input()
lst = list(n.split('-'))
answer = 0
for i in range(len(lst)):
    B = []
    B = list(lst[i].split('+'))
    for num in B:
        if i == 0:
            answer += int(num)

        else:
            answer -= int(num)

print(answer)

File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.java
import java.util.*;

public class Main {
  public static void main(String[] args){
    Scanner sc = new Scanner(System.in);

    String input = sc.nextLine();

    String[] series = input.split("\\-");

    int answer = 0;
    for (int i = 0; i < series.length; i++) {
      String[] last = series[i].split("\\+");
      for (String k : last) {
        if (i == 0) {
          answer += Integer.parseInt(k);
        } else {
          answer -= Integer.parseInt(k);
        }
      }
    }

    System.out.println(answer);
  }
}